File: test_login.py

import requests
import sys

session = requests.Session()

# 1. Get CSRF Cookie
try:
    print("Fetching CSRF cookie...")
    response = session.get('http://127.0.0.1:8000/sanctum/csrf-cookie')
    print(f"CSRF Status: {response.status_code}")
    print(f"Cookies: {session.cookies.get_dict()}")
    
    if 'XSRF-TOKEN' not in session.cookies:
        print("Error: XSRF-TOKEN cookie not found")
        sys.exit(1)

    xsrf_token = requests.utils.unquote(session.cookies['XSRF-TOKEN'])
    
    # 2. Login
    print("Attempting login...")
    headers = {
        'X-XSRF-TOKEN': xsrf_token,
        'Accept': 'application/json',
        'Referer': 'http://127.0.0.1:8000/login'
    }
    
    payload = {
        'email': 'admin@edoptcat.com',
        'password': 'password'
    }
    
    response = session.post('http://127.0.0.1:8000/login', json=payload, headers=headers)
    print(f"Login Status: {response.status_code}")
    print(f"Login Response: {response.text}")
    
    # 3. Check User
    print("Checking user...")
    response = session.get('http://127.0.0.1:8000/api/user', headers={'Accept': 'application/json'})
    print(f"User Status: {response.status_code}")
    print(f"User Response: {response.text}")

except Exception as e:
    print(f"An error occurred: {e}")

File: test_login.js

import axios from 'axios';
import { wrapper } from 'axios-cookiejar-support';
import { CookieJar } from 'tough-cookie';

const jar = new CookieJar();
const client = wrapper(axios.create({ jar }));

async function testLogin() {
    try {
        console.log('1. Getting CSRF cookie...');
        await client.get('http://127.0.0.1:8000/sanctum/csrf-cookie');

        // Extract XSRF-TOKEN from cookies
        const cookies = await jar.getCookies('http://127.0.0.1:8000');
        const xsrfToken = cookies.find(c => c.key === 'XSRF-TOKEN')?.value;

        if (!xsrfToken) {
            console.error('XSRF-TOKEN cookie not found');
            return;
        }

        console.log('XSRF-TOKEN found:', xsrfToken);

        console.log('1b. Initial Debug Session...');
        try {
            const debugInit = await client.get('http://127.0.0.1:8000/api/debug-session', {
                headers: {
                    'X-XSRF-TOKEN': decodeURIComponent(xsrfToken),
                    'Accept': 'application/json',
                    'Referer': 'http://127.0.0.1:8000/login',
                    'Origin': 'http://127.0.0.1:8000'
                }
            });
            console.log('Init Session ID:', debugInit.data.session_id);
            console.log('Init Config:', {
                driver: debugInit.data.config_session_driver,
                domain: debugInit.data.config_session_domain,
                secure: debugInit.data.config_session_secure // I didn't add this to api route but assumed defaults
            });
        } catch (e) { console.log('Init Debug failed:', e.message); }

        console.log('2. Logging in...');
        const loginResponse = await client.post('http://127.0.0.1:8000/login', {
            email: 'admin@edoptcat.com',
            password: 'password'
        }, {
            headers: {
                'X-XSRF-TOKEN': decodeURIComponent(xsrfToken),
                'Accept': 'application/json',
                'Referer': 'http://127.0.0.1:8000/login',
                'Origin': 'http://127.0.0.1:8000'
            }
        });

        console.log('Login Status:', loginResponse.status);

        console.log('3. Debug Session (Post-Login)...');
        try {
            const debugResponse = await client.get('http://127.0.0.1:8000/api/debug-session', {
                headers: {
                    'X-XSRF-TOKEN': decodeURIComponent(xsrfToken),
                    'Accept': 'application/json',
                    'Referer': 'http://127.0.0.1:8000/login',
                    'Origin': 'http://127.0.0.1:8000'
                }
            });
            console.log('Debug Status:', debugResponse.status);
            console.log('Debug Data:', debugResponse.data);
        } catch (e) {
            console.error('Debug session failed', e.message);
            if (e.response) console.error('Debug session status:', e.response.status);
        }

        console.log('4. Checking User...');
        const userResponse = await client.get('http://127.0.0.1:8000/api/user', {
            headers: {
                'X-XSRF-TOKEN': decodeURIComponent(xsrfToken),
                'Accept': 'application/json',
                'Referer': 'http://127.0.0.1:8000/login',
                'Origin': 'http://127.0.0.1:8000'
            }
        });
        console.log('User Status:', userResponse.status);
        console.log('User Role:', userResponse.data.role);

    } catch (error) {
        if (error.response) {
            console.error('Error Status:', error.response.status);
            console.error('Error Data:', error.response.data);
        } else {
            console.error('Error:', error.message);
        }
    }
}

testLogin();
